### ISSUE TITLE
What's the meaning of "stride" in FPN and  ResNet?

### ISSUE THREAD

**Issue by FeiZhuNiU** [NVIDIA/retinanet-examples#170]
```
class FPN(nn.Module):
    def __init__(self, features):
        super().__init__()

        self.stride = 128
```

```
class ResNet(vrn.ResNet):
    def __init__(self, layers=[3, 4, 6, 3], bottleneck=vrn.Bottleneck, outputs=[5], groups=1, width_per_group=64, url=None):
        self.stride = 128  
```

**Comment by FeiZhuNiU** [comment]
If it's the size rate between input size and each output layer, why it is a constant?

**Comment by yashnv** [comment]
what do you mean it is a constant? It is calculated from the simple formula: Input_size / pyramid level resolution, there are 5 pyramid levels: P3 to P7
3 means that P3 has resolution 2^3 lower than the input resolution. Does that answer you question?

**Comment by indigoviolet** [comment]
@yashnv :

> That stride is never used in code after initializing ResNet from torchvision.

It seems to be [used](https://github.com/NVIDIA/retinanet-examples/blob/db2acf36a91704088059f70d0a465289694150de/retinanet/train.py#L25)  to influence how the input images are [padded](https://github.com/NVIDIA/retinanet-examples/blob/db2acf36a91704088059f70d0a465289694150de/retinanet/data.py#L122).  Could you please elaborate on why 128 is chosen to pick this padding?

**Comment by yashnv** [comment]
The feature pyramid levels are in scales of 2^3, 2^4, ..., 2^7. The smallest downsampled level being the one corresponding to the last level, i.e., image input width/height being divided by 128. So 128 being used as stride in padding ensures that the input width/height to the network are divisible by 128.
